lib/apis/v3/node.go

```diff
@@ -98,6 +103,27 @@ type NodeWireguardSpec struct {
 	InterfaceIPv4Address string `json:"interfaceIPv4Address,omitempty" validate:"omitempty,ipv4"`
 }
 
+// NodeAddressType represents a node address type as a string
+type NodeAddressType string
+
+const (
+	NodeAddressIP                NodeAddressType = "IP"
+	NodeAddressVxlanTunnelIP     NodeAddressType = "VxlanTunnelIP"
+	NodeAddressVxlanMAC          NodeAddressType = "VxlanMAC"
+	NodeAddressIPIPTunnelIP      NodeAddressType = "IPIPTunnelIP"
+	NodeAddressWireguardTunnelIP NodeAddressType = "WireguardTunnelIP"
+	NodeAddressK8sInternalIP     NodeAddressType = "K8sInternalIP"
```

Comment:
I _think_ we have other fields that reference k8s that use `KubeFoo` rather than `K8sFoo` so probably best to use that